Lib/test/test_netrc.py

```diff
@@ -123,6 +123,19 @@ def test_security(self):
             os.chmod(fn, 0o622)
             self.assertRaises(netrc.NetrcParseError, netrc.netrc)
 
+    def test_file_not_found_in_home(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        with support.EnvironmentVarGuard() as environ:
+            environ.set('HOME', d)
```

Comment:
Serhiy is right that we need a test for this case and unless I'm missing something you can do it without doing any mocking:

```py
with support.EnvironmentVarGuard() as environ:
    environ.unset('HOME')
```